File: app/src/main/java/com/kickstarter/mock/factories/ActivityFactory.java
package com.kickstarter.mock.factories;

import com.kickstarter.models.Activity;

import org.joda.time.DateTime;

import androidx.annotation.NonNull;

public final class ActivityFactory {
  private ActivityFactory() {}

  public static Activity activity() {
    return Activity.builder()
      .category(Activity.CATEGORY_WATCH)
      .createdAt(new DateTime(123))
      .id(IdFactory.id())
      .updatedAt(new DateTime(456))
      .project(ProjectFactory.project())
      .user(UserFactory.user())
      .build();
  }

  public static @NonNull Activity friendBackingActivity() {
    return activity().toBuilder()
      .category(Activity.CATEGORY_BACKING)
      .build();
  }

  public static @NonNull Activity projectStateChangedActivity() {
    return activity().toBuilder()
      .category(Activity.CATEGORY_FAILURE)
      .project(ProjectFactory.failedProject())
      .build();
  }

  public static @NonNull Activity projectStateChangedPositiveActivity() {
    return activity().toBuilder()
      .category(Activity.CATEGORY_SUCCESS)
      .project(ProjectFactory.successfulProject())
      .build();
  }

  public static @NonNull Activity updateActivity() {
    return activity().toBuilder()
      .category(Activity.CATEGORY_UPDATE)
      .project(ProjectFactory.project())
      .user(UserFactory.user())
      .build();
  }
}

File: app/src/main/java/com/kickstarter/mock/factories/ActivityFactory.kt
package com.kickstarter.mock.factories

import com.kickstarter.mock.factories.ProjectFactory.failedProject
import com.kickstarter.mock.factories.ProjectFactory.project
import com.kickstarter.mock.factories.ProjectFactory.successfulProject
import com.kickstarter.mock.factories.UserFactory.user
import com.kickstarter.models.Activity
import com.kickstarter.models.Activity.Companion.builder
import org.joda.time.DateTime

object ActivityFactory {
    @JvmStatic
    fun activity(): Activity {
        return builder()
            .category(Activity.CATEGORY_WATCH)
            .createdAt(DateTime(123))
            .id(IdFactory.id().toLong())
            .updatedAt(DateTime(456))
            .project(project())
            .user(user())
            .build()
    }

    @JvmStatic
    fun friendBackingActivity(): Activity {
        return activity().toBuilder()
            .category(Activity.CATEGORY_BACKING)
            .build()
    }

    @JvmStatic
    fun projectStateChangedActivity(): Activity {
        return activity().toBuilder()
            .category(Activity.CATEGORY_FAILURE)
            .project(failedProject())
            .build()
    }

    @JvmStatic
    fun projectStateChangedPositiveActivity(): Activity {
        return activity().toBuilder()
            .category(Activity.CATEGORY_SUCCESS)
            .project(successfulProject())
            .build()
    }

    @JvmStatic
    fun updateActivity(): Activity {
        return activity().toBuilder()
            .category(Activity.CATEGORY_UPDATE)
            .project(project())
            .user(user())
            .build()
    }
}
